### PR TITLE
build ACE Armadillo and Boost

### PR DESCRIPTION
**Pull request by paskino**
test CI

### CODE REVIEW

**Comment by paskino** [comment]
Problem 1: `cblas.h` is not found.

`cblas.h` is part of [`libatlas-dev`](https://packages.ubuntu.com/bionic/amd64/libatlas-base-dev/filelist), however it can be seen that the file has the name `cblas-atlas.h`. 

In my Ubuntu 18.04 which I use to test the SuperBuild these builds don't fail and I have the following files:

```
ls -lh /usr/include/x86_64-linux-gnu/
total 316K
-rw-r--r-- 1 root root 4.3K Jan 24 12:53 a.out.h
drwxr-xr-x 2 root root  12K Feb 22 06:00 asm
drwxr-xr-x 2 root root  12K Jan 24  2020 atlas
drwxr-xr-x 3 root root  20K Mar  2 06:43 bits
drwxr-xr-x 4 root root 4.0K Jan 25  2021 c++
-rw-r--r-- 1 root root  34K Sep 13  2017 cblas-atlas.h
-rw-r--r-- 1 root root  21K Jun 17  2017 cblas_f77.h
lrwxrwxrwx 1 root root   42 Jan 24  2020 cblas.h -> /etc/alternatives/cblas.h-x86_64-linux-gnu
-rw-r--r-- 1 root root  444 Nov 28  2017 cblas_mangling.h
-rw-r--r-- 1 root root  30K Nov 28  2017 cblas-netlib.h
-rw-r--r-- 1 root root  45K Sep 19  2017 cblas-openblas.h
-rw-r--r-- 1 root root 9.5K Sep 13  2017 clapack.h
-rw-r--r-- 1 root root 3.5K Feb 17 23:38 expat_config.h
-rw-r--r-- 1 root root  47K Sep 19  2017 f77blas.h
-rw-r--r-- 1 root root 3.5K Jan 24 12:53 fpu_control.h
drwxr-xr-x 2 root root 4.0K Mar  2 06:43 gnu
-rw-r--r-- 1 root root 4.8K Jan 24 12:53 ieee754.h
-rw-r--r-- 1 root root 2.2K Jun  4  2020 jconfig.h
drwxr-xr-x 2 root root 4.0K Jan 24  2020 libxslt
-rw-r--r-- 1 root root 4.7K Sep 19  2017 openblas_config.h
drwxr-xr-x 2 root root 4.0K Mar  5  2021 python2.7
drwxr-xr-x 2 root root 4.0K Dec 17 06:37 python3.6m
drwxr-xr-x 2 root root  12K Mar  2 06:43 sys
```

`cblas.h` is a symbolic link to a file that seems to have been created by `update-alternatives`. 
Looking into the file I get 

```
#ifndef CBLAS_H
#define CBLAS_H

#include <stddef.h>
#include "openblas_config.h"
...
```
That means that this `cblas.h` is part of OpenBLAS, and `diff` confirms that.

**Comment by KrisThielemans** [comment]
I did see somewhere a recommendation to install `openblas`.

**Comment by paskino** [comment]
[Error ](https://github.com/SyneRBI/SIRF-SuperBuild/runs/5445683784?check_suite_focus=true#step:7:29487)building `siemens_to_ismrmrd`

```
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:39:22: error: expected primary-expression before ‘const’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
                      ^~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:39:2: error: ‘BOOST_FOREACH’ was not declared in this scope
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:39:2: note: suggested alternative: ‘BOOST_FINAL’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
  BOOST_FINAL
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp: In member function ‘std::__cxx11::string XProtocol::getXMLString::operator()(const XProtocol::XNodeParamMap&) const’:
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:98:22: error: expected primary-expression before ‘const’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
                      ^~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:98:2: error: ‘BOOST_FOREACH’ was not declared in this scope
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:98:2: note: suggested alternative: ‘BOOST_FINAL’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
  BOOST_FINAL
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp: In member function ‘std::__cxx11::string XProtocol::getXMLString::operator()(const XProtocol::XNodeParamArray&) const’:
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:115:22: error: expected primary-expression before ‘const’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
                      ^~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:115:2: error: ‘BOOST_FOREACH’ was not declared in this scope
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:115:2: note: suggested alternative: ‘BOOST_FINAL’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
  BOOST_FINAL
```

**Comment by KrisThielemans** [comment]
this seems to get into a nightmare. What about forgetting about building BOOST ourselves? Installing that is easy enough on all platforms. We can sort that out later.

Same for ACE and Armadillo I guess, but possibly easier.

**Comment by paskino** [comment]
In fact, the last build [without ](https://github.com/SyneRBI/SIRF-SuperBuild/runs/5452484862?check_suite_focus=true)`siemens_to_ismrmrd` has completed correctly while building ACE, Boost and Armadillo.

**Comment by paskino** [comment]
See also https://github.com/ismrmrd/siemens_to_ismrmrd/issues/92

File: version_config.cmake [hunk 1]
```diff
@@ -179,15 +182,12 @@ if (DEVEL_BUILD)
 
   ## siemens_to_ismrmrd
   set(DEFAULT_siemens_to_ismrmrd_URL https://github.com/ismrmrd/siemens_to_ismrmrd )
-  set(DEFAULT_siemens_to_ismrmrd_TAG 6d0ab3d3d0c8ade5c0526db1c6af9825008425ad)
+  set(DEFAULT_siemens_to_ismrmrd_TAG origin/master)
```
Comment (KrisThielemans) [inline version_config.cmake:185]: this will come and bite us. I'd prefer to fix a version here

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
still ISMRMRD 1.5.0 at https://github.com/SyneRBI/SIRF-SuperBuild/blob/8afee138baf6989bbf7770a687437ba52862eb91/version_config.cmake#L111

**Comment by paskino** [comment]
I think this is ready for re-review. I will remove the references to my branch and merge if approved.
One of the CI config failed at install prerequisite phase.

File: docker/user_sirf-ubuntu.sh [hunk 2]
```diff
@@ -3,10 +3,10 @@
 set -ev
 INSTALL_DIR="${1:-/opt}"
 # set default URL/tag
-: ${SIRF_SB_URL=https://github.com/SyneRBI/SIRF-SuperBuild}
-: ${SIRF_SB_TAG=master}
+: ${SIRF_SB_URL=https://github.com/paskino/SIRF-SuperBuild}
+: ${SIRF_SB_TAG=check_build_build_boost}
```
Comment (KrisThielemans) [inline docker/user_sirf-ubuntu.sh:7]: I don't know why you modified this in this PR as `user_sirf-ubuntu.sh` isn't used on GHA (only in docker and the VM), but will have to revert it as you say.

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
looks great. tiny comment

File: CHANGES.md [hunk 3]
```diff
@@ -9,12 +9,12 @@
 - Boost: fix cases where the wrong version of boost could be found [#627](https://github.com/SyneRBI/SIRF-SuperBuild/issues/627)
 - add the CERN ROOT library (if USE_ROOT=ON, but defaults to OFF), which can be used by STIR to read GATE ROOT files.
 - updated versions:
-  - ISMRMRD: 1.5.0
-  - siemens_to_ismrmrd: [6d0ab3d3d0c8ade5c0526db1c6af9825008425ad](https://github.com/ismrmrd/siemens_to_ismrmrd/commit/6d0ab3d3d0c8ade5c0526db1c6af9825008425ad)
+  - ISMRMRD: 1.7.0
+  - siemens_to_ismrmrd: [b87759e49e53dab4939147eb52b7a0e6465f3d04](https://github.com/ismrmrd/siemens_to_ismrmrd/commit/b87759e49e53dab4939147eb52b7a0e6465f3d04)
```
Comment (KrisThielemans) [inline CHANGES.md:13]: probably helpful to say `> 1.7.0 with bug-fix for boost/foreach.hpp`

File: docker/user_sirf-ubuntu.sh (hunk 2 above)
Comment (paskino) [reply docker/user_sirf-ubuntu.sh:7]: because I'm confused. And it's me who created the GHA (initially)! Probably it went out of hand, certainly.

**Review by KrisThielemans — APPROVED** [review]
squash merge please

**Pull request merged**